hubblestack/extmods/hubble_mods/command_line_parser.py

```diff
@@ -0,0 +1,273 @@
+
+    delimiter = runner_utils.get_param_for_module(block_id, block_dict, 'delimiter')
+    if not delimiter:
+        error['delimiter'] = 'No delimiter provided'
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+def execute(block_id, block_dict, chain_args=None):
+    """
+    Execute the module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param chain_args:
+        Chained argument dictionary, (If any)
+        Example: {'result': "/some/path/file.txt", 'status': True}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing command_line_parser module for id: {0}'.format(block_id))
+
+    try:
+        chained_param = runner_utils.get_chained_param(chain_args)
+        command_line = chained_param.get('cmdline')
+
+        key_aliases = runner_utils.get_param_for_module(block_id, block_dict, 'key_aliases')
+        delimiter = runner_utils.get_param_for_module(block_id, block_dict, 'delimiter')
+
+        log.debug("command_line: {0}, key_aliases: {1}, delimeter: {2}".format(command_line, key_aliases, delimiter))
+        ret_match_list = []
+
+        for key_alias in key_aliases:
+            log.debug("looping with key_alias %s", key_alias)
+            key_alias = key_alias.lstrip("-")
+
+            if key_alias not in command_line:
+                log.info("key_alias %s not found in command line %s", key_alias, command_line)
+                continue
+
+            regex_base = "".join(["(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}", key_alias, "\"{0,1}\'{0,1}\s*", delimiter])
+            regex_list = []
+            regex_pattern = "".join([regex_base, "\s*openingbracket.*closingbracket(?=(\s|$))"])
+            braces_list = [('\[','\]'), ('\{','\}'), ('\(','\)')]
+            for item in braces_list:
+                regex = re.sub("openingbracket", item[0], regex_pattern)
+                regex = re.sub("closingbracket", item[1], regex)
+                regex_list.append(regex)
+            regex1 = "".join([regex_base, "\s*([\",\']).*?\\2"])
+            regex2 = "".join([regex_base, "\s*.+?(?=(\s|$))"])
+            regex_list.append(regex1)
+            regex_list.append(regex2)
+
+            for regex in regex_list:
+                log.debug("looping with regex : %s", regex)
+                match_list = _get_match_list(regex, key_alias, command_line, delimiter)
+                if match_list:
+                    ret_match_list.extend(match_list)
+                    break
+
+        return runner_utils.prepare_positive_result_for_module(block_id, ret_match_list)
+
+    except Exception as e:
+        log.exception("Some exception occurred in command_line_parser's parse_cmdline function %s", e)
+        return runner_utils.prepare_negative_result_for_module(block_id, None)
```

Comment:
We should return an error message in negative result instead of None.

Reply:
This was as per old module result.